tests/RobotFramework/tests/pkcs11/private_key_storage.robot

```diff
@@ -53,6 +58,26 @@ Select Private key using PKCS#11 URI
     Set tedge-p11-server Uri    value=pkcs11:token=tedge;object=tedge
     Tedge Reconnect Should Succeed
 
+    Set tedge-p11-server Uri    value=
+
+Select Private key using a request URI
+    [Documentation]    Like above, we select the key using a URI, but this time we include it in a request, which means
+    ...    we can select different keys without restarting tedge-p11-server.
+
+    Set tedge-p11-server Uri    value=
+    Tedge Reconnect Should Succeed
+
+    Execute Command    softhsm2-util --init-token --free --label token10 --pin "123456" --so-pin "123456"
+    Execute Command    cmd=tedge config set device.key_uri pkcs11:token=token10
+    Tedge Reconnect Should Fail With    Failed to find a signing key
```

Comment:
This surprises me a little that without the `object=tedge` component of the pkcs11 uri, that they key is not found.

Reply:
It might be more obvious to the reader by using an incorrect token name rather than just leaving the `object=tedge`.

```
tedge config set device.key_uri pkcs11:token=token123
```

Reply:
The logic is that if `token` attribute is present, we select that token. Only then we try to find any keys on it (in this case there are none).

The docs section mentions this although I realize now that it doesn't sound very clear:

> It is also possible to use a URI that identifies a token in settings like device.key_uri. The URI will then be used to select a token, but the key will be selected automatically, though the selected key may be wrong if there are multiple to choose from. [1]

[1]: https://thin-edge.github.io/thin-edge.io/references/hsm-support/#key-selection

i.e. after selecting the token, we automatically try to find a matching key **on that token**.

Reply:
Though if there is only one key for the given token, wouldn't that be enough to find it?

Reply:
In this case, we create a new empty token, with no keys in it, which, looking back, is indeed unnecessary, because we can just use an incorrect token name.
Switched to just using a non-existent token in a21f01bb.

Reply:
Yeah that makes the test easier to read